public/API/API_GetGameList.php

```diff
@@ -31,22 +33,52 @@
 
 $withAchievements = (bool) request()->query('f');
 $withHashes = (bool) request()->query('h');
+$offset = (int) request()->query('o');
+$count = (int) request()->query('c');
 
-getGamesListByDev(null, $consoleID, $dataOut, 1, false, $withAchievements ? 0 : 2);
+$query = DB::table('GameData')
+    ->leftjoin('Console AS c', 'c.ID', '=', 'GameData.ConsoleID')
+    ->leftjoin('Achievements AS ach', 'ach.GameID', '=', 'GameData.ID')
+    ->leftjoin('LeaderboardDef AS ld', 'ld.GameID', '=', 'GameData.ID')
+    ->select(
+        'GameData.*',
+        'c.Name as ConsoleName',
+        DB::raw('MAX(ach.DateModified) as DateModified'),
+        DB::raw('COALESCE(GameData.achievements_published,0) AS NumAchievements'),
+        DB::raw('COUNT(DISTINCT ld.ID) AS NumLBs')
+    )
+    ->where('ConsoleID', $consoleID);
+
+if ($withAchievements) {
+    $query
+        ->where('achievements_published', '>', 0);
+}
+
+if ($offset > 0 && $count > 0) {
+    $query
+        ->offset($offset)
+        ->take($count);
+}
+
+$queryResponse = $query
+    ->groupBy('GameData.ID')
+    ->orderBy('GameData.Title', 'asc')
+    ->get();
 
 $response = [];
-foreach ($dataOut as &$entry) {
+
+foreach ($queryResponse as $game) {
     $responseEntry = [
-        'Title' => $entry['Title'],
-        'ID' => $entry['ID'],
-        'ConsoleID' => $entry['ConsoleID'],
-        'ConsoleName' => $entry['ConsoleName'],
-        'ImageIcon' => $entry['GameIcon'],
-        'NumAchievements' => $entry['NumAchievements'] ?? 0,
-        'NumLeaderboards' => $entry['NumLBs'] ?? 0,
-        'Points' => $entry['MaxPointsAvailable'] ?? 0,
-        'DateModified' => $entry['DateModified'],
-        'ForumTopicID' => $entry['ForumTopicID'],
+        'Title' => $game->Title,
+        'ID' => $game->ID,
+        'ConsoleID' => $game->ConsoleID,
+        'ConsoleName' => $game->ConsoleName,
+        'ImageIcon' => $game->ImageIcon,
+        'NumAchievements' => $game->NumAchievements ?? 0,
+        'NumLeaderboards' => $game->NumLBs ?? 0,
+        'Points' => $game->points_total ?? 0,
+        'DateModified' => $game->DateModified,
+        'ForumTopicID' => $game->ForumTopicID,
     ];
     $responseEntry['NumAchievements'] = (int) $responseEntry['NumAchievements'];
```

Comment:
These three lines doing the casts can be removed now that the query is using Eloquent ORM.

The cast for NumAchievements is still needed and can be lifted up:
```php
$responseEntry = [
    'Title' => $game->Title,
    'ID' => $game->ID,
    'ConsoleID' => $game->ConsoleID,
    'ConsoleName' => $game->ConsoleName,
    'ImageIcon' => $game->ImageIcon,
    'NumAchievements' => (int) $game->NumAchievements ?? 0,
    'NumLeaderboards' => $game->NumLBs ?? 0,
    'Points' => $game->points_total ?? 0,
    'DateModified' => $game->DateModified,
    'ForumTopicID' => $game->ForumTopicID,
];
```